src/components/AdminDashboard.jsx

```diff
@@ -43,21 +46,34 @@ class AdminDashboard extends React.Component {
   }
 
   render() {
+    const { location, children, params } = this.props
+    const { roles } = this.props.data.currentUser
+
+    // HACK: Setting params.adminPerms helps us hide non-supervolunteer functionality
+    params.adminPerms = hasRole('ADMIN', roles || [])
```

Comment:
This is an icky icky hack, but drastically simplifies the implementation here.  I'm not sure if there is a way to do this better, but if not, I think it would be worth the hackiness to keep the rest of the code cleaner.